pkg/fakerp/fakerp.go

```diff
@@ -24,7 +27,76 @@ import (
 	"github.com/openshift/openshift-azure/pkg/util/vault"
 )
 
-func GetDeployer(log *logrus.Entry, cs *api.OpenShiftManagedCluster) api.DeployFn {
+func debugDeployerError(ctx context.Context, log *logrus.Entry, cs *api.OpenShiftManagedCluster, err error, testConfig api.TestConfig) error {
+	authorizer, err := azureclient.GetAuthorizerFromContext(ctx, api.ContextKeyClientAuthorizer)
+	if err != nil {
+		return err
+	}
+
+	deploymentOperations := azureclient.NewDeploymentOperationsClient(ctx, cs.Properties.AzProfile.SubscriptionID, authorizer)
+
+	operations, err := deploymentOperations.List(ctx, cs.Properties.AzProfile.ResourceGroup, "azuredeploy", nil)
+	if err != nil {
+		log.Warnf("failed to get deployment operations: %v", err)
+		return err
+	}
+
+	for _, op := range operations {
+		if *op.Properties.ProvisioningState == "Succeeded" {
+			continue
+		}
+
+		b, _ := json.MarshalIndent(op, "", "  ")
+		log.Debug(string(b))
+
+		if testConfig.ArtifactDir != "" &&
+			*op.Properties.TargetResource.ResourceType == "Microsoft.Compute/virtualMachineScaleSets" {
+			s, err := newSSHer(ctx, cs)
+			if err != nil {
+				log.Warnf("newSSHer failed: %v", err)
+				continue
+			}
+
+			for _, app := range cs.Properties.AgentPoolProfiles {
+				prefix := names.GetScalesetName(&app, "")
+				if !strings.HasPrefix(*op.Properties.TargetResource.ResourceName, prefix) {
+					continue
+				}
+
+				for i := int64(0); i < app.Count; i++ {
+					hostname := *op.Properties.TargetResource.ResourceName + fmt.Sprintf("%06s", strconv.FormatInt(i, 36))
+					cli, err := s.Dial(ctx, hostname)
+					if err != nil {
+						log.Warnf("Dial failed: %v", err)
+						continue
+					}
+
+					err = s.RunRemoteCommandAndSaveToFile(cli, "sudo journalctl", testConfig.ArtifactDir+"/"+hostname+"-early-journal")
+					if err != nil {
+						log.Warnf("RunRemoteCommandAndSaveToFile failed: %v", err)
```

Comment:
i feel this warning should be done inside `RunRemoteCommandAndSaveToFile` instead of repeating yourself, but may clash with keeping that func relatively clean :man_shrugging: wdyt?